Realm/RLMProperty.mm

```diff
@@ -343,6 +343,9 @@ - (instancetype)initSwiftPropertyWithName:(NSString *)name
     }
     else if ([_objcRawType isEqualToString:@"@\"NSNumber\""]) {
         const char *numberType = [propertyValue objCType];
+        if (!numberType) {
+            @throw RLMException(@"Can't persist NSNumber directly: use a Swift-native number type or provide a default value, which allows to infer the type");
```

Comment:
It's intentional. There aren't really any good ways to let the user specify the type (can't get the generic type at runtime, can't use the protocol hack we use in obj-c, and using a subclass means that you have to have casts at runtime to assign NSNumbers to the property (and it disables the use of tagged pointers)).